pallene/checker.lua

```diff
@@ -226,15 +225,15 @@ local letrec_groups = {
 }
 
 function Checker:check_program(prog_ast)
-
+    local tls = prog_ast.tls
```

Comment:
Please add an `assert(prog_ast._tag == "ast.Progra.Program")` before this line. Generally speking, whenever you have a piece of code that assumes that the tag must be a certain type, we should add an assert to help catch it.

For example, if someone passes a list of toplevel nodes instead of a prog_ast then this assignment will silently succeed, returning nil... The error will only be caught further down when we do an ipairs on it.